src/transformState.js

```diff
@@ -4,8 +4,24 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.map((key) => delete state[key]);
+        break;
+
+      default:
+        Object.keys(state).map((key) => delete state[key]);
```

Comment:
what if someone pass invalid action type?
all properties will be deleted?)
pass this to `clear` case